library/std/src/primitive_docs.rs

```diff
@@ -499,10 +517,38 @@ mod prim_unit {}
 #[stable(feature = "rust1", since = "1.0.0")]
 mod prim_pointer {}
 
+#[doc(primitive = "array")]
+#[doc(alias = "[")]
+#[doc(alias = "]")]
 #[doc(alias = "[]")]
+#[doc(alias = "[ ]")]
 #[doc(alias = "[T;N]")] // unfortunately, rustdoc doesn't have fuzzy search for aliases
 #[doc(alias = "[T; N]")]
-#[doc(primitive = "array")]
+#[doc(alias = "[T")] // unfortunately, rustdoc doesn't find prefixes for aliases
+#[doc(alias = "[T;")]
+#[doc(alias = "[T;N")]
+#[doc(alias = "[T; N")]
+#[doc(alias = "[;]")]
+#[doc(alias = "[ ;]")]
+#[doc(alias = "[; ]")]
+#[doc(alias = "[ ; ]")]
+#[doc(alias = "[;")]
+#[doc(alias = "[ ;")]
+#[doc(alias = "[T;0]")]
+#[doc(alias = "[T; 0]")]
+#[doc(alias = "[T;0")]
+#[doc(alias = "[T; 0")]
+#[doc(alias = "[T;1]")]
+#[doc(alias = "[T; 1]")]
+#[doc(alias = "[T;1")]
+#[doc(alias = "[T; 1")]
+#[doc(alias = ";")]
+#[doc(alias = ";0")]
+#[doc(alias = "; 0")]
+#[doc(alias = ";1")]
+#[doc(alias = "; 1")]
+#[doc(alias = ";N")]
+#[doc(alias = "; N")]
```

Comment:
Rather than adding a whole bunch of these, would you be interested in adding fuzzy search support for doc aliases? That seems more useful, and less maintenance for the standard library.

Reply:
Eh… I’d have no idea how to do something like that. That search is probably JavaScript? I never really used JavaScript before… And also I’m not sure if “full” fuzzy search for non-symbolic doc aliases is even wanted in the general case. E.g. if I type “`in`”, I’m not sure if I’d want to see the “inject” → “Iterator::fold” alias in my search results.

Reply:
I do admit, that this list may have become a bit lengthy. Feel free to point out anything that goes “too far” :-D

(also, for this reason the PR is split up into many individual commits, so the “overkill” changes – if any – can be named and called out easily)

Reply:
I also figured out (after the fact) that `#[doc(alias = "…", alias = "…", …)]` lists are a thing, so if the number of lines should be reduced, I could group them.

Reply:
Iirc the search already does fuzzy searching (based on levenshtein distance), maybe that just needs to be extended to aliases or something if it doesn't already cover these cases.

Reply:
Search lives around here IIRC: https://github.com/rust-lang/rust/blob/c5799b2a73bd03d1d7b4a432fa0a0cb93e7288db/src/librustdoc/html/static/js/search.js#L79